### ISSUE TITLE
Spam Mitigation 

### ISSUE THREAD

**Issue by vuong177** [Pylons-tech/pylons#583]
## Problem Definition

- Pylons has no-fee transactions, so we need to create a prevent spam tx mechanism. 
 
## Proposal

   - An account only can send a limited number of transactions in one block ( or a period ).
   - Number of transactions that can be sent is a param. This param could modify by on-chain government.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by MikeSofaer** [comment]
Makes sense to me. I think we also need to gate account creation with some kind of voucher system, sort of like fee grants. The simplest way to start is Firebase's App Check: https://firebase.google.com/docs/app-check

Having the chain validate an App Check token during account creation should be relatively simple, and work a lot like the Stripe and IAP vouchers, and will be very strong, I think.

We could also have a governance flag to use App Check for every transaction if things start getting attacked.

**Comment by faddat** [comment]
@vuong177 -- I think this is a great first pass. May want to check in with @faisalnaveedRNS and-- sir, thank you.

**Comment by faddat** [comment]
Maybe we define some kind of order of implmentation?

1) Vuong's concept (frankly I am very interested to see how he accomplishes this.  Better than my own concepts, tho vulnerable to consistent abuse at exactly the limit)
2) App-check: this is going to be a very strange chain.  No one is doing this.  Love it.  this also complements Vuong's concept.

@faisalnaveedRNS -- I'd also like to know if you have thoughts on additional layers here.  Currently I'd like to see Vuong's built first due to simplicity, but I also want to track this issue here.

**Comment by vuong177** [comment]
@faddat @MikeSofaer  my implementation in that [branch](https://github.com/Pylons-tech/pylons/tree/spam-mitigation) 
I'm going to write proto files. Please take a look!
